static/src/javascripts/lib/robust.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+/*
+    Swallows (and reports) exceptions. Designed to wrap around modules at the "bootstrap" level.
+    For example "comments throwing an exception should not stop auto refresh"
+ */
+
+import reportError from 'lib/report-error';
+
+function catchErrors(fn: Function): ?Error {
+    let error;
+
+    try {
+        fn();
+    } catch (e) {
+        error = e;
+    }
+
+    return error;
+}
+
+function log(name: string, error: Error, customReporter?: Function): void {
+    let reporter = reportError;
+
+    if (customReporter) {
+        reporter = customReporter;
+    }
+
+    if (window.console && window.console.warn) {
+        window.console.warn('Caught error.', error.stack);
+    }
+
+    reporter(error, { module: name }, false);
+}
+
+function catchErrorsAndLog(name: string, fn: Function, reporter?: Function) {
+    const error = catchErrors(fn);
+
+    if (error) {
+        log(name, error, reporter);
+    }
+}
+
+function catchErrorsAndLogAll(modules: Array<any>) {
+    return modules.map(pair => {
+        const [name, fn] = pair;
+        catchErrorsAndLog(name, fn);
+        return undefined;
```

Comment:
Unfortunately we have to stick with `map` here, since this is called within `forEach` loops, which rely on returning an array. I tried, if I can pass the output of the actual function through, but they are somehow inconsistent ...

Do you have a better idea?